### PR TITLE
added widget support in form items

### LINKED ISSUE #338: Support widget for form item label and checkbox trailing/leading text

### PR DESCRIPTION
**Pull request by sagardspeed2**
## Describe your changes
Added widget support feature in form items

## Issue ticket number and link

Closes #
closes #338 

## Checklist before requesting a review

- [x] I have performed a self-review of my code
- [ ] I have added tests
- [ ] I have added a changeset `pnpm changeset add`
- [x] I have added example usage in the kitchen sink app

### CODE REVIEW

**Review by evshi — COMMENTED** [review]
@sagardspeed2 per the ticket can you also add widget support for the label attribute?

File: packages/runtime/src/widgets/Form/Checkbox.tsx [hunk 1]
```diff
@@ -36,6 +46,7 @@ export const CheckboxWidget: React.FC<CheckBoxProps> = (props) => {
 
   return (
     <EnsembleFormItem valuePropName="checked" values={values}>
+      {leadingContent}
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/Checkbox.tsx:49]: Does this break the form submission?

Comment (sagardspeed2) [reply packages/runtime/src/widgets/Form/Checkbox.tsx:49]: Yup, when we used the leadingContent, the checkbox checked stat is not updating.
so if we are not using the leadingContent, should I remove the leadingText prop from the definition ?

File: packages/runtime/src/widgets/Form/Checkbox.tsx [hunk 2]
```diff
@@ -37,7 +50,7 @@ export const CheckboxWidget: React.FC<CheckBoxProps> = (props) => {
             : undefined),
         }}
       >
-        {values?.trailingText}
+        {trailingContent}
```
Comment (evshi) [inline packages/runtime/src/widgets/Form/Checkbox.tsx:53]: Trailing and leading text are mutually exclusive so we can support either or and use direction: LTR/RTL to make it appear before and after

**Review by evshi — APPROVED** [review]

**Pull request merged**